cirq/ops/pauli_string_test.py

```diff
@@ -532,6 +533,21 @@ def test_pow():
     assert (1j * p)**-1 == -1j * p
 
 
+def test_rpow():
+    a, b = cirq.LineQubit.range(2)
+
+    u = cirq.unitary(np.exp(1j * np.pi / 2 * cirq.Z(a) * cirq.Z(b)))
+    np.testing.assert_allclose(u, np.diag([1j, -1j, -1j, 1j]), atol=1e-8)
+
+    u = cirq.unitary(np.exp(-1j * np.pi / 4 * cirq.Z(a) * cirq.Z(b)))
+    cirq.testing.assert_allclose_up_to_global_phase(u,
+                                                    np.diag([1, 1j, 1j, 1]),
+                                                    atol=1e-8)
```

Comment:
Stronger assertion:

```python
cirq.testing. assert_allclose(u, np.diag(np.sqrt(-1j), np.sqrt(1j), np.sqrt(1j), np.sqrt(-1j)), atol=1e-8)
```

Reply:
I intentionally only did it up to global phase to make the test simpler to write, to make sure the complications weren't hiding a mistake.